imap/mboxname.c

```diff
@@ -234,10 +234,14 @@ EXPORTED int mboxname_lock(const char *mboxname, struct mboxlock **mboxlockptr,
                      nonblock, fname);
     if (!r) lockitem->l.locktype = locktype;
     else if (errno == EWOULDBLOCK) r = IMAP_MAILBOX_LOCKED;
-    else r = errno;
+    else r = IMAP_IOERROR;
 
 done:
-    if (r) remove_lockitem(lockitem);
+    if (r) {
+        xsyslog(LOG_ERR, "can not lock mailbox",
+                "name=%s error=%s (%m)", mboxname, error_message(r));
```

Comment:
The format string in xsyslog should use angle brackets for output arguments, e.g. `name=<%s>`, that's also the format that xsyslog uses for its internal output.
Also, the `%m` format is a glibc-extension, and it doesn't look necessary to me because xsyslog outputs the latest errno in its `syserror` log variable for LOG_ERR anyway.

Reply:
Fixed